platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/DelegateServletFilter.java

```diff
@@ -14,92 +14,82 @@
+import org.codice.ddf.platform.filter.SecurityFilter;
+import org.ops4j.pax.web.extender.whiteboard.ExtenderConstants;
+import org.ops4j.pax.web.extender.whiteboard.internal.util.ServicePropertiesUtils;
 import org.osgi.framework.Bundle;
 import org.osgi.framework.BundleContext;
 import org.osgi.framework.Constants;
 import org.osgi.framework.FrameworkUtil;
-import org.osgi.framework.InvalidSyntaxException;
 import org.osgi.framework.ServiceReference;
+import org.osgi.service.http.whiteboard.HttpWhiteboardConstants;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * {@link DelegateServletFilter} is meant to detect any ServletFilters
- * and call their {@link FilterChain}. If none are found it becomes a pass
- * through filter.
- *
+ * {@link DelegateServletFilter} is meant to detect any {@link SecurityFilter}s and doFilter through
+ * those before continuing on its {@link FilterChain}. If no {@link SecurityFilter}s are found, it
+ * becomes a pass-through filter.
  */
 public class DelegateServletFilter implements Filter {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(DelegateServletFilter.class);
 
-    private FilterConfig filterConfig;
+    final TreeSet<ServiceReference<SecurityFilter>> sortedSecurityFilterServiceReferences;
 
-    protected BundleContext getContext() {
+    /**
+     * Indicates that the {@link SecurityFilter}s in
+     * {@link DelegateServletFilter#sortedSecurityFilterServiceReferences} have all been
+     * initialized
+     */
+    private Boolean filtersHaveBeenInitialized;
+
+    FilterConfig filterConfig;
+
+    BundleContext getContext() {
         Bundle cxfBundle = FrameworkUtil.getBundle(DelegateServletFilter.class);
         if (cxfBundle != null) {
             return cxfBundle.getBundleContext();
         }
         return null;
     }
 
+    public DelegateServletFilter() {
+        sortedSecurityFilterServiceReferences = new TreeSet<>();
+        filtersHaveBeenInitialized = false;
+    }
+
     @Override
     public void destroy() {
+        // do nothing
     }
 
     @Override
     public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
             FilterChain filterChain) throws IOException, ServletException {
-
-        Collection<ServiceReference<Filter>> referenceCollection = null;
-        BundleContext context = getContext();
-
-        if (null != context) {
-            try {
-                referenceCollection = context.getServiceReferences(Filter.class, null);
-            } catch (InvalidSyntaxException ise) {
-                LOGGER.debug("Should not get this exception as there is no filter being passed.");
-            }
-        }
-
-        if (referenceCollection != null && !referenceCollection.isEmpty()) {
-            LOGGER.debug("Found {} filter(s), now filtering...", referenceCollection.size());
-
+        if (!CollectionUtils.isEmpty(sortedSecurityFilterServiceReferences)) {
```

Comment:
After looking at this some more, I believe there is an issue with hanging onto the service references in this way. Just like the bundle contexts, service references can also go stale. I believe that is why we were grabbing them and sorting for every request.